### PR TITLE
Callback rework

### PR DESCRIPTION
**Pull request by lgebhardt**
Reworks the operations processor callbacks to use ActiveSupport callbacks.
Adds callbacks to resource operations.
Reworks create_has_many_link to create_has_many_links.

### CODE REVIEW

File: lib/jsonapi/resource.rb [hunk 1]
```diff
@@ -1,49 +1,111 @@
...
+      end
+    end
+
     def remove
+      run_callbacks :remove do
+        _remove
+      end
+    end
+
+    def _remove
```
Comment (dgeb) [inline lib/jsonapi/resource.rb]: `_remove` and all the other underscored methods should be private.

**Comment by barelyknown** [comment]
I sent a PR to the `callback_rework` branch with `README` documentation. https://github.com/cerebris/jsonapi-resources/pull/66

**Comment by lgebhardt** [comment]
@dgeb, I moved the new methods down into the private section. The other underscored methods that are public need to stay that way.

**Comment by dgeb** [comment]
@lgebhardt thanks for making those updates. Looks like there's still some public / private labeling and indentation confusion.

**Comment by dgeb** [comment]
Great work @lgebhardt and @barelyknown! :shipit:

**Pull request merged**